src/Components/EmailNotificationForm/EmailNotificationForm.js

```diff
@@ -37,28 +43,48 @@ class EmailNotificationForm extends Component {
         email: this.state.email,
         language: this.state.language,
       }
-      // this.props.postNewEmailSubscriber(newEmailSubscriber);
+
+      postNewEmailSubscriber(newEmailSubscriber)
+        .then(response => {
+          this.setState({ isSubmitting: true });
+          if (response.status !== 200) {
+            throw new Error (`${response.status} : Sorry, cannot fetch the data.`)
+          }
+          if (!response.ok) {
+            throw new Error ('Something has gone wrong, please try again.')
+          }
+          return response.json()
+        })
+        .then(message => {
+          this.setState({ isSubmitting: false });
+          this.setState({ successMessage: message.success });
+        })
+        .catch(error => {
+          this.setState({ isSubmitting: false });
+          this.setState({ error: error.message });
```

Comment:
What do you think about refactoring this to one line `this.setState({ isSubmitting: false,  error: error.message })` ?